packages/bolt-interactions/src/interaction-flow.spec.ts

```diff
@@ -0,0 +1,145 @@
+import JestReceiver from '@slack-wrench/jest-bolt-receiver';
+import { App, ConversationStore, MemoryStore } from '@slack/bolt';
+import delay from 'delay';
+
+import { InteractionFlow, interactionFlow } from './index';
+
+const random = 'RANDOM';
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+InteractionFlow.interactionIdGenerator = () => random;
+
+describe('Bolt interaction flows', () => {
+  const flowName = 'test';
+  const flowId = InteractionFlow.createFlowId(flowName, random);
+
+  let store: ConversationStore;
+  let receiver: JestReceiver;
+  let app: App;
+  let state: any;
+
+  beforeEach(() => {
+    store = new MemoryStore();
+    InteractionFlow.store = store;
+    receiver = new JestReceiver();
+    state = { foo: 'bar' };
+
+    app = new App({ receiver, token: 'TOKEN' });
+  });
+
+  it('prevents duplicate flow names', () => {
+    expect.assertions(1);
+
+    interactionFlow(flowName, noop)(app);
+
+    expect(() => interactionFlow(flowName, noop)(app)).toThrow(
+      /declared twice/,
+    );
+  });
+
+  it('Allows state to be set from commands', async () => {
+    expect.assertions(2);
+
+    const command = '/command';
+
+    interactionFlow(flowName, flow => {
+      app.command(command, async ({ ack }) => {
```

Comment:
I switched to mixing `flow` and `app` commands as I expect we'll need to start flows outside of slack interactions. This is a touch more flexible.